packages/Config/RectorConfig.php

```diff
@@ -165,9 +165,18 @@ public function ruleWithConfiguration(string $rectorClass, array $configuration)
 
         $this->singleton($rectorClass);
         $this->afterResolving($rectorClass, static function (ConfigurableRectorInterface $configurableRector) use (
-            $configuration
+            $configuration,
+            $rectorClass
         ): void {
+            $configuration = array_merge(
+                SimpleParameterProvider::provideArrayParameter(Option::RECTOR_CONFIGURATION)[$rectorClass] ?? [],
+                $configuration
+            );
             $configurableRector->configure($configuration);
+
+            SimpleParameterProvider::addParameter(Option::RECTOR_CONFIGURATION, [
+                $rectorClass => $configuration,
```

Comment:
The `$rectorClass` as configurable index is essential here, to be re-used to merge with existing service config on next `->ruleWithConfiguration()` called.